python/arb.cpp

```diff
@@ -0,0 +1,235 @@
+#include <vector>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include <domain_decomposition.hpp>
+#include <model.hpp>
+#include <load_balance.hpp>
+#include <profiling/meter_manager.hpp>
+#include <recipe.hpp>
+#include <rss_cell.hpp>
+
+#include "print.hpp"
+#include "recipe.hpp"
+
+namespace pb = pybind11;
+
+template <typename T>
+arb::util::any wrap_any(T value) {
+    return arb::util::any(std::move(value));
+}
+
+struct spike_recorder {
+    using f = arb::model::spike_export_function;
+    using spike_vec = std::vector<arb::spike>;
+    spike_vec spike_store;
+
+    f callback() {
+        return [&](const spike_vec& spikes) {
+            spike_store.insert(spike_store.end(), spikes.begin(), spikes.end());
+        };
+    }
+};
+
+std::unique_ptr<spike_recorder> make_spike_recorder(arb::model& m) {
+    auto r = arb::util::make_unique<spike_recorder>();
+    m.set_global_spike_callback(r->callback());
+    return r;
+}
+
```

Comment:
The C++ symbols should probably have static linkage or be in their own (or an anonymous, or the `arb`) namespace.

Reply:
Sure. I just put them here as a POC of how such a convenience function could be written to help Python users.